pkg/scheduler/dependency_graph.go

```diff
@@ -336,29 +366,44 @@ func (rl *ReplicaList) Swap(i, j int) {
 	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
 }
 
-func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
-	label := labels.Set{"flow": flow.Name}
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]client.Replica, []client.Replica, error) {
```

Comment:
allocate != create. It allocate replicas for the algorithm for different purposes. How would you call the function otherwise? Will address everything else